src/models/ipdCourse.js

```diff
@@ -0,0 +1,60 @@
+const { Model } = require('sequelize');
+
+export default (sequelize, DataTypes) => {
+  class IpdCourse extends Model {
+    static associate(models) {
+      // IpdCourse.belongsTo(models.Objective, { foreignKey: 'objectiveId', as: 'objective' });
+      // eslint-disable-next-line max-len
+      // IpdCourse.belongsTo(models.ActivityReportObjective, { foreignKey: 'activityReportObjectiveId', as: 'activityReportObjective' });
+
+      IpdCourse.hasMany(models.IpdCourse, {
+        foreignKey: 'mapsTo',
+        as: 'mapsFromIpdCourse',
+      });
+      IpdCourse.belongsTo(models.IpdCourse, {
+        foreignKey: 'mapsTo',
+        as: 'mapsToIpdCourse',
+      });
+    }
+  }
+  IpdCourse.init({
+    id: {
+      type: DataTypes.INTEGER,
+      primaryKey: true,
+      autoIncrement: true,
+      allowNull: false,
+    },
+    name: {
+      type: DataTypes.TEXT,
+      allowNull: false,
+    },
+    mapsTo: {
+      type: DataTypes.INTEGER,
+      allowNull: true,
+      references: {
+        model: {
+          tableName: 'ValidFor',
```

Comment:
This is a copy past error, should be the current tables name

Reply:
oops nice catch. Still a wip (draft).